src/main/scala/mesosphere/marathon/raml/ConstraintConversion.scala

```diff
@@ -47,7 +47,7 @@ trait ConstraintConversion {
       case Protos.Constraint.Operator.UNLIKE => ConstraintOperator.Unlike
       case Protos.Constraint.Operator.MAX_PER => ConstraintOperator.MaxPer
     }
-    Constraint(c.getField, operator, Option(c.getValue))
+    Constraint(c.getField, operator, Option(c.getValue).map(_.trim).filter(_.nonEmpty))
```

Comment:
how is this change related to this PR? did leak from another PR or debugging session?

Reply:
Since the default protobuf value for strings is empty string, we have to manually check and filter them to produce a correct result.

Reply:
are we using proto3? i thought we were still using proto2 and could rely on hasXXX methods to determine the presence of a field

Reply:
According to the [spec](https://developers.google.com/protocol-buffers/docs/proto#optional) for proto2:
> If the default value is not specified for an optional element, a type-specific default value is used instead: for strings, the default value is the empty string.